scanpointgenerator/generators/linegenerator.py

```diff
@@ -68,30 +72,35 @@ def __init__(self, name, units, start, stop, num, alternate_direction=False):
 
         self.axes = self.name  # For GDA
 
-    def iterator(self):
-
-        for i in range_(self.num):
-            point = Point()
-
-            for axis_index in range_(self.num_axes):
-                axis_name = self.name[axis_index]
-                start = self.start[axis_index]
-                step = self.step[axis_index]
-
-                point.positions[axis_name] = start + i * step
-                point.lower[axis_name] = start + (i - 0.5) * step
-                point.upper[axis_name] = start + (i + 0.5) * step
-
-            point.indexes = [i]
-            yield point
+    def produce_points(self):
+        self.points = {}
+        self.bounds = {}
+        for axis in range_(self.num_axes):
+            axis_name = self.name[axis]
+            start = self.start[axis]
+            stop = self.stop[axis]
+            d = stop - start
+            if self.num == 1:
+                self.points[axis_name] = np.array([start])
```

Comment:
It's a significant departure from the regular case where `start` and `stop` are members of `self.points` (and `self.bounds` extends slightly past them).